homeassistant/components/homematicip_cloud/light.py

```diff
@@ -117,3 +124,143 @@ def supported_features(self):
     async def async_turn_off(self, **kwargs):
         """Turn the light off."""
         await self._device.set_dim_level(0)
+
+
+class HomematicipNotificationLight(HomematicipGenericDevice, Light):
+    """Representation of HomematicIP Cloud dimmer light device."""
+
+    _channel_index = None
+
+    # Dictionary to translate between RGBColorState and hs_color
+    _color_switcher = None
+
+    def __init__(self, home, device, channel_index):
+        """Initialize the dimmer light device."""
+        self._channel_index = channel_index
+        if self._channel_index == 2:
+            super().__init__(home, device, 'Top')
+        else:
+            super().__init__(home, device, 'Bottom')
+
+        from homematicip.base.enums import RGBColorState
+        self._color_switcher = {
+            RGBColorState.WHITE: [0.0, 0.0],
+            RGBColorState.RED: [0.0, 100.0],
+            RGBColorState.YELLOW: [60.0, 100.0],
+            RGBColorState.GREEN: [120.0, 100.0],
+            RGBColorState.TURQUOISE: [180.0, 100.0],
+            RGBColorState.BLUE: [240.0, 100.0],
+            RGBColorState.PURPLE: [300.0, 100.0]
+        }
+
+    @property
+    def _channel(self):
+        return self._device.functionalChannels[self._channel_index]
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._channel.dimLevel > 0.0
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..255."""
+        return int(self._channel.dimLevel * 255)
+
+    @property
+    def hs_color(self):
+        """Return the hue and saturation color value [float, float]."""
+        simple_rgb_color = self._channel.simpleRGBColorState
+        return self._color_switcher.get(simple_rgb_color, [0.0, 0.0])
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the generic device."""
+        attr = super().device_state_attributes
+        if self.is_on:
+            attr.update({
+                ATTR_COLOR_NAME:
+                    self._channel.simpleRGBColorState
+            })
+        return attr
+
+    @property
+    def name(self):
+        """Return the name of the generic device."""
+        return "{} {}".format(super().name, 'Notification')
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_BRIGHTNESS | SUPPORT_COLOR
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return "{}_{}_{}".format(self.__class__.__name__,
+                                 self.post,
+                                 self._device.id)
+
+    async def async_turn_on(self, **kwargs):
+        """Turn the light on."""
+        # Use hs_color from kwargs,
+        # if not applicable use current hs_color.
+        hs_color = kwargs.get(ATTR_HS_COLOR, None)
```

Comment:
Put `self.hs_color` as default instead of `None`.

Reply:
Done